styx-common/src/main/java/com/spotify/styx/model/Workflow.java

```diff
@@ -41,6 +41,11 @@
   @JsonProperty
   public abstract Optional<URI> componentUri();
 
+  // this gives a hint that the workflow registration/modification came from Styx API
+  // instead of any schedule source
+  @JsonProperty("__from_api")
+  public abstract Optional<String> fromApi();
```

Comment:
String seems like a suboptimal type for the reasons I laid out in my other comment. But maybe it's more convenient for serialization compared to for example Enums.